contracts/dummy_application.sol

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+pragma solidity ^0.5.0;
+
+import "./zecale_client_application.sol";
+
+// Trivial dummy application used to test the Zecale protocol. Transactions
+// demonstrate knowledge of the multiplicative inverse of scalar value. The
+// contract state records the set of scalars seen so far, rejecting
+// transactions for scalars that have already been seen.
+contract DummyApplication is ZecaleClientApplication
```

Comment:
`ZecaleClientApplication` -> `ZecaleBaseApplication` (the notion of BaseApplication is used in the paper, and "client" induces that there's a "server" which I find confusing for on-chain applications)

Reply:
(Sorry - I missed this one)
Would one of these work?
- `ZecaleApplicationBase`
- `ZecaleApplication` or
- `IZecaleApplication`?

(`ZecaleBaseApplication` sounds like it is part of the hierarchy of Zecale itself, i.e. "the Application class from which `Zecale` inherits", whereas here we mean the base class / interface for all Zecale applications.)

Reply:
`IZecaleApplication` works well (I like the convention to prepend `I` to indicate that this is an interface as also done in the python code)

Reply:
renamed